js/app/controllers/editorcontroller.js

```diff
@@ -79,6 +79,43 @@ app.controller('EditorController', ['$scope', 'TimezoneService', 'AutoCompletion
 		$scope.registerPostHook = function(callback) {
 			$scope.postEditingHooks.push(callback);
 		};
+		
+		$scope.parse_summary = function() {
+			var summary = $scope.properties.summary.value;
+			
+			// Parse text to extract schedule from string beginning
+			var timePattern = "(\\d{1,2}:\\d{2}(?:\\s?[ap]\\.?m\\.?)?)";
+			var linkPattern = "\\s?[-/]\\s?";
+			var scheduleExp = new RegExp("^"+timePattern+"(?:"+linkPattern+timePattern+")?\\s", 'i');
+			var matches = scheduleExp.exec(summary);
+			
+			if(matches !== null) {
+		
+			  	// Strip schedule string from summary
+			  	summary = summary.substring(matches.shift().length);
+			
+			  	var start = matches[0];
+			  	var end = matches[1];
+				
+				// Convert string to time
+				var schemes = ["HH:mm", "hh:mm a"];
+				start = moment(start, schemes);
+				if(typeof end !== 'undefined') {
+					end = moment(end, schemes);
+				}
+				else {
+					end = start.clone().add(1, 'hour');
+				}
+				
+				// Update event properties
+				$scope.properties.dtstart.value = moment($scope.properties.dtstart.value.set(
```

Comment:
I would have preferred to directly update the objects `$scope.properties.dtstart/dtend.value` using their [set](http://momentjs.com/docs/#/get-set/set/) method rather creating new ones, but this wouldn't work. I don't understand why ...